core/ledger/confighistory/mgr_test.go

```diff
@@ -300,100 +296,210 @@ func newTestEnvForSnapshot(t *testing.T) *testEnvForSnapshot {
-	require.NoError(t, dbHandle.writeBatch(batch, true))
+		batch, err = prepareDBBatch(
+			map[string]*peer.CollectionConfigPackage{
+				"chaincode1": cc1CollConfigPackageNew,
+				"chaincode2": cc2CollConfigPackageNew,
+				"chaincode3": cc3CollConfigPackageNew,
+			},
+			100,
+		)
+		require.NoError(t, err)
+		require.NoError(t, db.writeBatch(batch, true))
 
-	fileHashes, err = env.retriever.ExportConfigHistory(env.testSnapshotDir, testNewHashFunc)
-	require.NoError(t, err)
+		cc1configBytes, err := proto.Marshal(cc1CollConfigPackage)
+		require.NoError(t, err)
+		cc2configBytes, err := proto.Marshal(cc2CollConfigPackage)
+		require.NoError(t, err)
+		cc3configBytes, err := proto.Marshal(cc3CollConfigPackage)
+		require.NoError(t, err)
+		cc1configBytesNew, err := proto.Marshal(cc1CollConfigPackageNew)
+		require.NoError(t, err)
+		cc2configBytesNew, err := proto.Marshal(cc2CollConfigPackageNew)
+		require.NoError(t, err)
+		cc3configBytesNew, err := proto.Marshal(cc3CollConfigPackageNew)
+		require.NoError(t, err)
 
-	cc1configBytesNew, err := proto.Marshal(cc1collConfigPackageNew)
-	require.NoError(t, err)
-	cc2configBytesNew, err := proto.Marshal(cc2collConfigPackageNew)
-	require.NoError(t, err)
-	cc3configBytesNew, err := proto.Marshal(cc3collConfigPackageNew)
-	require.NoError(t, err)
-	expectedCollectionConfigs = []*compositeKV{
-		{&compositeKey{ns: "lscc", key: "chaincode1~collection", blockNum: 100}, cc1configBytesNew},
-		{&compositeKey{ns: "lscc", key: "chaincode1~collection", blockNum: 50}, cc1configBytes},
-		{&compositeKey{ns: "lscc", key: "chaincode2~collection", blockNum: 100}, cc2configBytesNew},
-		{&compositeKey{ns: "lscc", key: "chaincode2~collection", blockNum: 50}, cc2configBytes},
-		{&compositeKey{ns: "lscc", key: "chaincode3~collection", blockNum: 100}, cc3configBytesNew},
-		{&compositeKey{ns: "lscc", key: "chaincode3~collection", blockNum: 50}, cc3configBytes},
+		storedKVs := []*compositeKV{
+			{&compositeKey{ns: "lscc", key: "chaincode1~collection", blockNum: 100}, cc1configBytesNew},
+			{&compositeKey{ns: "lscc", key: "chaincode1~collection", blockNum: 50}, cc1configBytes},
+			{&compositeKey{ns: "lscc", key: "chaincode2~collection", blockNum: 100}, cc2configBytesNew},
+			{&compositeKey{ns: "lscc", key: "chaincode2~collection", blockNum: 50}, cc2configBytes},
+			{&compositeKey{ns: "lscc", key: "chaincode3~collection", blockNum: 100}, cc3configBytesNew},
+			{&compositeKey{ns: "lscc", key: "chaincode3~collection", blockNum: 50}, cc3configBytes},
+		}
+
+		return storedKVs, ccConfigInfo
 	}
-	verifyExportedConfigHistory(t, env.testSnapshotDir, fileHashes, expectedCollectionConfigs)
+
+	cleanup := func() {
+		require.NoError(t, os.RemoveAll(filepath.Join(env.testSnapshotDir, snapshotDataFileName)))
+		require.NoError(t, os.RemoveAll(filepath.Join(env.testSnapshotDir, snapshotMetadataFileName)))
+	}
+
+	t.Run("confighistory is empty", func(t *testing.T) {
+		retriever := env.mgr.GetRetriever("ledger1", nil)
```

Comment:
Not sure if this could cause a flake... can you change "ledger1" to "ledger0". As, the subtests are expected to be independent, better to make sure that even some reordering does not affect these.

Reply:
Also, in the similar lines, better to call `cleanup()` in `defer` as opposed to at the end. Else, one failed could cascade to others and the purpose of subtests (as opposed to a single monolithic) is defeated.

Reply:
I will make both changes. It wouldn't cause a flake because `t.Run()` is executed serially one after the other unless we specify `t.Parallel()` inside the `func()` present in the `t.Run()`.

Reply:
Not sure what does `t.Parallel()` has to do with my comment. What I mentioned is that I am not certain if the ordering of the subtests is guaranteed? If yes, then what order is guaranteed - in the order they appear in the file or in some other order (like sort order on the name of the subtest). Even if this is guaranteed, it would be fragile such that if you swap the subtest order in the code, it would break the tests

Reply:
Agree on independence and rearrangement. My response was mainly related to the flakes issue that you raised. The `t.Run()` is blocked call and it runs in the same order it occurs in the main test. Only If we put `t.Parallel()`, the execution would be made unblocked and sub-tests can interleave. This is what I wanted to convey to mention that it wouldn't result in a flake.